src/function.py

```diff
@@ -34,18 +36,18 @@ def zip_and_upload_folder(client: CogniteClient, folder: Path, file_name: str) -
     raise FunctionDeployError("Failed to upload file to Cognite Data Fusion")
 
 
-def await_function_deployment(client: CogniteClient, external_id: str, wait_time_seconds: int):
+def await_function_deployment(client: CogniteClient, external_id: str, wait_time_seconds: int) -> Optional[Function]:
     t_end = time.time() + wait_time_seconds
-    while time.time() < t_end:
+    while time.time() <= t_end:
         function: Optional[Function] = client.functions.retrieve(external_id=external_id)
         if function is not None:
             if function.status == "Ready":
-                return True
+                return function
             if function.status == "Failed":
                 raise FunctionDeployError(function.error["trace"])
-        time.sleep(3.0)
+        time.sleep(randint(3, 15))  # nosec
```

Comment:
@pavelzubarev any particular reason for the randomness?

Reply:
Distribute the load if many instances are waiting... but that brakes test, actually...